docs/testing/testing.adoc

```diff
@@ -0,0 +1,225 @@
+[[testing]]
+= Testing
+
+We offer `org.neo4j.springframework.boot.test.autoconfigure.data.DataNeo4jTest` and `org.neo4j.springframework.boot.test.autoconfigure.data.ReactiveDataNeo4jTest`
+inside an additional module under the following coordinates:
+`org.neo4j.springframework.data:spring-data-neo4j-rx-spring-boot-test-autoconfigure`. Include the following dependency in your project setup
+
+[source,xml,subs="verbatim,attributes"]
+.spring-data-neo4j-rx-spring-boot-test-autoconfigure for Maven
+----
+<dependency>
+    <groupId>org.neo4j.springframework.data</groupId>
+    <artifactId>spring-data-neo4j-rx-spring-boot-test-autoconfigure</artifactId>
+    <version>{spring-data-neo4j-rx-version}</version>
+    <scope>test</scope>
+</dependency>
+----
+
+Or with Gradle
+
+[source,groovy,subs="verbatim,attributes"]
+.spring-data-neo4j-rx-spring-boot-test-autoconfigure for Gradle
+----
+dependencies {
+    testImplementation 'org.neo4j.springframework.data:spring-data-neo4j-rx-spring-boot-test-autoconfigure:{spring-data-neo4j-rx-version}'
+}
+----
+
+Both `@DataNeo4jTest` and `@ReactiveDataNeo4jTest` are Spring Boot https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-testing[test slices].
+By default, they provide the Neo4j test harness through the transitive dependency of https://github.com/neo4j/neo4j-java-driver-spring-boot-starter/tree/master/neo4j-java-driver-test-harness-spring-boot-autoconfigure[neo4j-java-driver-test-harness-spring-boot-autoconfigure].
+Both slices provides all the necessary infrastructure for tests using Neo4j: A driver bean, a transaction manager, a client, a template and declared repositories,
+in their imperative or reactive variants.
+`@DataNeo4jTest` provides both variants if reactive repositories are enabled while `@ReactiveDataNeo4jTest` provides only reactive infrastructure.
+
+The dependencies include the Neo4j test harness in version 3.5.x by default. Thus, reactive database access and multiple databases cannot be tested out of the box.
+The reason for doing this is simple: Spring Data's JDK baseline is JDK 8. The Neo4j test harness in version 4 requires JDK 11+, so we cannot make it the default.
+
+Here are the available options. Both test slices include `@ExtendWith(SpringExtension.class)` so that they run automatically with JUnit 5 (JUnit Jupiter).
```

Comment:
```suggestion
Here are the available options.
Both test slices include `@ExtendWith(SpringExtension.class)` so that they run automatically with JUnit 5 (JUnit Jupiter).
```